src/colorizer/Dataset.ts

```diff
@@ -550,6 +552,11 @@ export default class Dataset {
   }
 
   public buildTrack(trackId: number): Track {
+    const cachedTrack = this.cachedTracks.get(trackId);
```

Comment:
this is a cool idea. Were you needing to dynamically compute tracks frequently?

Reply:
Yes! This replaces an optimization I was making in the Annotation List View, where I have to build out every track that's been annotated to create its thumbnail view. Building a track takes O(N) time where N is the number of objects in the dataset, and becomes inefficient if you need to build lots of tracks.

(Arguably we could do a single pass through the dataset on load to create all tracks at once. It would take O(Nlog(N)) since we also need to sort every track chronologically, and also would increase the memory footprint of a Dataset since we'd be storing copies of the time/track/ID arrays.)

Reply:
I imagine that for large datasets, users rarely would touch every track, so I think the lazy version is pretty good